pkg/codegen/internal/test/testdata/other-owned/schema.json

```diff
@@ -0,0 +1,275 @@
+        }
+      },
+      "required": ["foo"],
+      "language": {
+        "nodejs": {
+          "requiredInputs": []
+        }
+      },
+      "type": "object"
+    }
+  },
+  "resources": {
+    "example::Resource": {
+      "properties": {
+        "bar": {
+          "type": "string",
+          "secret": true
+        }
+      },
+      "inputProperties": {
+        "bar": {
+          "type": "string",
+          "secret": true
+        }
+      },
+      "type": "object"
+    },
+    "example::OtherResource": {
+      "isComponent": true,
+      "properties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "inputProperties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "type": "object"
+    },
+    "bar::BarResource": {
+      "isComponent": true,
+      "properties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "inputProperties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "type": "object"
+    },
+    "foo::FooResource": {
+      "isComponent": true,
+      "properties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "inputProperties": {
+        "foo": {
+          "$ref": "#/resources/example::Resource"
+        }
+      },
+      "type": "object"
+    },
+
+    "example::OverlayResource": {
+      "isOverlay": true,
+      "properties": {
+        "foo": {
+          "$ref": "#/types/example::ConfigMapOverlay"
+        },
+        "bar": {
+          "$ref": "#/types/example::EnumOverlay"
+        }
+      },
+      "inputProperties": {
+        "foo": {
+          "$ref": "#/types/example::ConfigMapOverlay"
+        },
+        "bar": {
+          "$ref": "#/types/example::EnumOverlay"
+        }
+      },
+      "type": "object"
+    },
+    "example::TypeUses": {
+      "properties": {
+        "foo": {
+          "$ref": "#/types/example::Object"
+        },
+        "bar": {
+          "$ref": "#/types/example::SomeOtherObject"
+        },
+        "baz": {
+          "$ref": "#/types/example::ObjectWithNodeOptionalInputs"
+        }
+      },
+      "inputProperties": {
+        "foo": {
+          "$ref": "#/types/example::Object"
+        },
+        "bar": {
+          "$ref": "#/types/example::SomeOtherObject"
+        },
+        "baz": {
+          "$ref": "#/types/example::ObjectWithNodeOptionalInputs"
+        }
+      },
+      "type": "object"
+    }
+  },
+  "functions": {
+    "example::argFunction": {
+      "inputs": {
+        "properties": {
+          "arg1": {
+            "$ref": "#/resources/example::Resource"
+          }
+        }
+      },
+      "outputs": {
+        "properties": {
+          "result": {
+            "$ref": "#/resources/example::Resource"
+          }
+        }
+      }
+    },
+    "example::overlayFunction": {
```

Comment:
How do we decide what to include into this test case's schema beyond a basic case? I guess you copied it from some other test case. How did you pick one? Does every object in this schema test something meaningful?